src/puppetlabs/ezbake/dependency_utils.clj

```diff
@@ -30,6 +30,47 @@
       (classpath/merge-versions-from-managed-coords
        (:managed-dependencies lein-project))))
 
+(defn snapshot-version?
+  [version]
+  ;; dependencies managed by lein-parent have nil versions so guard against that
+  (boolean (and version (re-find #"SNAPSHOT" version))))
+
+(defn get-full-snapshot-version
+  [lein-project coords]
+  (let [full-version (-> (aether/resolve-artifacts
+                           :coordinates [coords]
+                           :repositories (:repositories lein-project)
+                           :mirrors (:mirrors lein-project)
+                           :local-repo (:local-repo lein-project))
+                       first
+                       meta
+                       :result
+                       .getArtifact
+                       .getVersion)]
+    (if-not (re-find #"SNAPSHOT$" full-version)
+      full-version
+      (throw
+        (Exception.
```

Comment:
I looked through the standard Exception subclasses and couldn't see anything that looked appropriate. It's not an IllegalArgumentException since the problem isn't the argument, it's that no versions matching the snapshot have been deployed. I suppose I could change it to `IllegalStateException`, which is accurate enough if you squint and handwave a bit, in order to make it easier to catch just this problem.